rotkehlchen/api/rest.py

```diff
@@ -4092,3 +4092,45 @@ def download_user_db_snapshot(self, timestamp: Timestamp) -> Response:
                 wrap_in_fail_result('No file was found'),
                 status_code=HTTPStatus.NOT_FOUND,
             )
+
+    def _query_addresses(
+        self,
+        addresses: List[ChecksumEthAddress],
+    ) -> Dict[ChecksumEthAddress, Optional[str]]:
+        addr_mappings = {}
+        for addr in addresses:
+            new_name = self.rotkehlchen.chain_manager.ethereum.ens_reverse_lookup(addr)
+            addr_mappings[addr] = new_name
+            self.rotkehlchen.data.db.add_ens_mapping(addr, new_name)
```

Comment:
This code makes `n` node queries where `n` is the number of addresses. It could be only one if `_ens_reverse_lookup` used from `rotkehlchen.chain.ethereum.utils.multicall`. This would require to move the for loop logic to the `_ens_reverse_lookup` but now you make only one node query. Don't know what @LefterisJP and you think about this idea

Reply:
Hm, how fast a call is performed when using multicall? Can it give an ability to query all addresses each time a page is loaded? 
This data isn't supposed to be queried often (mostly only when user adds a new wallet) but yeah, it could be useful for example when app is updated (and user already have a lot of connected wallets) to query data for all wallets faster. @LefterisJP what do you think?